connectors/src/connectors/intercom/temporal/activities.ts

```diff
@@ -210,3 +228,213 @@ export async function syncCollectionActivity({
     currentSyncMs,
   });
 }
+
+/**
+ * This activity is responsible for retrieving the list
+ * of team ids to sync for a given connector.
+ */
+export async function getTeamIdsToSyncActivity(connectorId: ModelId) {
+  const teams = await IntercomTeam.findAll({
+    attributes: ["teamId"],
+    where: {
+      connectorId: connectorId,
+    },
+  });
+  return teams.map((t) => t.teamId);
+}
+
+/**
+ * This activity is responsible for syncing the conversations of a given Team.
+ * If the team is not allowed anymore, it will delete all its data.
+ * If the team is not present on Intercom anymore, it will delete all its data.
+ * If the team is present on Intercom and is allowed, it will sync its conversations.
+ */
+export async function syncTeamOnlyActivity({
+  connectorId,
+  teamId,
+  currentSyncMs,
+}: {
+  connectorId: ModelId;
+  teamId: string;
+  currentSyncMs: number;
+}): Promise<boolean> {
+  const connector = await _getIntercomConnectorOrRaise(connectorId);
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  const loggerArgs = {
+    workspaceId: dataSourceConfig.workspaceId,
+    connectorId,
+    provider: "intercom",
+    dataSourceName: dataSourceConfig.dataSourceName,
+  };
+
+  const teamOnDB = await IntercomTeam.findOne({
+    where: {
+      connectorId,
+      teamId,
+    },
+  });
+  if (!teamOnDB) {
+    logger.error({ loggerArgs, teamId }, "[Intercom] Team not found");
+    return false;
+  }
+
+  // If our rights were revoked we delete the team and its conversations
+  if (teamOnDB.permission === "none") {
+    await deleteTeamAndConversations({
+      connectorId,
+      dataSourceConfig,
+      team: teamOnDB,
+    });
+    return false;
+  }
+
+  // If the team does not exists on Intercom we delete the team and its conversations
+  const teamOnIntercom = await fetchIntercomTeam(
+    connector.connectionId,
+    teamId
+  );
+  if (!teamOnIntercom) {
+    await deleteTeamAndConversations({
+      connectorId,
+      dataSourceConfig,
+      team: teamOnDB,
+    });
+    return false;
+  }
+
+  // Otherwise we update the team name and lastUpsertedTs
+  await teamOnDB.update({
```

Comment:
I'm not sure I understand -- where is the upsert actually happening ?

Reply:
Ok I guess team by itself doesn't have any datasource upserts, it's basically just its name that is kept in sync ?

Reply:
Yes exactly! I hesitated to keep the column in this table as we don't have a datasource for a Team, it's only a folder, but at the same time I think it can be helpful to know when it was last refreshed!